tests/Unit/Utilities/PregTest.php

```diff
@@ -0,0 +1,282 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Pelago\Emogrifier\Tests\Unit\Utilities;
+
+use Pelago\Emogrifier\Utilities\Preg;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @covers \Pelago\Emogrifier\Utilities\Preg
+ */
+final class PregTest extends TestCase
+{
+    /**
+     * @var string|array<string>
+     */
+    private $replaceCallbackReplacement;
+
+    /**
+     * @var ?string
```

Comment:
The `?` notation only works for native type declarations. For PHPDoc type annotations, we need to use `|null`:

```suggestion
     * @var string|null
```

Reply:
The `?` notation works with PHPStan - without it, it generates an error when the property is assigned to `null`.